sdk-java/src/main/java/io/littlehorse/sdk/wfsdk/WfRunVariable.java

```diff
@@ -1,53 +1,44 @@
      */
     WfRunVariable jsonPath(String path);
 
     /**
-     * Enables the storage of variables with a Non-null {@link IndexType}. For enhanced efficiency,
-     * it offers two types of indexing: Remote Index and Local Index. {@link IndexType#REMOTE_INDEX
-     * Remote}: This type of indexing is recommended for variables with low cardinality, which means
-     * they have relatively few distinct values. For example, storing userId. {@link
-     * IndexType#LOCAL_INDEX Local}: Local Index is designed for variables with high cardinality.
-     *
-     * @param indexType Defines Local or Remote Index
-     * @return same {@link WfRunVariable} instance
+     * Marks the variable as "Required", meaning that the ThreadSpec cannot be
+     * started without this variable being provided as input. For Entrypoint
+     * ThreadSpec's, this also triggers the WfSpec Required Variable Compatibiltiy
+     * Rules.
+     * @return
      */
-    WfRunVariable withIndex(@NonNull IndexType indexType);
+    WfRunVariable required();
 
     /**
-     * Enables the storage of specific attributes inside a Json Variable. For enhanced efficiency,
-     * it offers two types of indexing: Remote Index and Local Index. {@link IndexType#REMOTE_INDEX
-     * Remote}: This type of indexing is recommended for variables with low cardinality, which means
-     * they have relatively few distinct values. For example, storing userId. {@link
-     * IndexType#LOCAL_INDEX Local}: Local Index is designed for variables with high cardinality.
-     *
-     * @param indexType Defines Local or Remote Index
-     * @param jsonPath Json Attribute path starting with $. e.g: $.userId
+     * Marks the Variable as "Searchable", which:
+     * - Creates an Index on the Variable in the LH Data Store
+     * - Due to the fact that the Variable is now Searchable, all future WfSpec
+     *   versions must use the same Type for this Variable.
      * @return same {@link WfRunVariable} instance
-     * @throws io.littlehorse.sdk.common.exception.LHMisconfigurationException if jsonPath doesn't
-     *     start with $. or adding a jsonIndex to a non-json variable
      */
-    WfRunVariable withJsonIndex(@NonNull String jsonPath, @NonNull IndexType indexType);
+    WfRunVariable searchable();
 
     /**
-     * Makes this variable persistent across WfSpec versions. This enables searching for
-     * this `Variable` without specifying a WfSpec Version, and it also requires that all
-     * future versions of the WfSpec have the same variable with
-     * @return
+     * Marks the JSON_OBJ or JSON_ARR Variable as "Searchable", and creates an
+     * index on the specified field.
+     * @param fieldPath is the JSON Path to the field that we are indexing.
+     * @param fieldType is the type of the field we are indexing.
+     * @return same {@link WfRunVariable} instance
      */
-    WfRunVariable persistent();
+    WfRunVariable searchableOn(String fieldPath, VariableType fieldType);
```

Comment:
I'm wondering if we do a validation when we are indexing a Getable to make sure that the json attribute type matches with this.

Reply:
We will do that. The question is how do we handle when validation fails. I don't think it should be an error that stops the workflow. We need a way to send a "warning" to the user.

Reply:
We will do that. The question is how do we handle when validation fails. I don't think it should be an error that stops the workflow. We need a way to send a "warning" to the user.